test-grapesy/Test/Sanity/Exception.hs

```diff
@@ -0,0 +1,196 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans   #-}
+
+-- | Handling of exceptions occurring in an RPC on a shared connection.
+--
+-- These tests check for the behavior described in
+-- <https://github.com/well-typed/grapesy/issues/102>. In particular, there are
+-- two conditions that should hold when an exception occurs in the scope of a
+-- call (on either the server or client, e.g. inside either 'mkRpcHandler' or
+-- 'withRPC'):
+--
+-- 1. Other ongoing calls on that connection are not terminated, and
+-- 2. future calls are still possible.
+module Test.Sanity.Exception where
```

Comment:
This comment seems to describe only the client side of these tests?

Reply:
True, I'll add the server side properties